synapse/handlers/cas_handler.py

```diff
@@ -201,33 +193,63 @@ async def handle_ticket(
             args["redirectUrl"] = client_redirect_url
         if session:
             args["session"] = session
-        username, user_display_name = await self._validate_ticket(ticket, args)
+        username, attributes = await self._validate_ticket(ticket, args)
+
+        # first check if we're doing a UIA
+        if session:
+            return await self._sso_handler.complete_sso_ui_auth_request(
+                self._auth_provider_id, username, session, request,
+            )
+
+        # otherwise, we're handling a login request.
+
+        # Ensure that the attributes of the logged in user meet the required
+        # attributes.
```

Comment:
Unfortunately this isn't quite the same as the SAML checks so I'm not quite sure it could be abstracted.